### PR TITLE
chore: Add release commits

### PR DESCRIPTION
**Pull request by JPZ13**
These are commits that we've used to get the release process for the fork of argo-workflows working. We will squash these into one and note for future releases

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -120,7 +120,7 @@ jobs:
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v2
         with:
-          version: v0.10.4
+          version: v0.9.1
```
Comment (tico24) [inline .github/workflows/release.yaml]: We should use the later version.

File: .github/workflows/changelog.yaml [hunk 2]
```diff
@@ -30,4 +30,4 @@ jobs:
           title: 'docs: updated CHANGELOG.md'
           commit-message: 'docs: updated CHANGELOG.md'
           branch: create-pull-request/changelog
-          signoff: true
\ No newline at end of file
+          signoff: true
```
Comment (tico24) [inline .github/workflows/changelog.yaml]: Change not needed?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -150,97 +150,78 @@ jobs:
           PLATFORM: ${{ matrix.platform }}
           TARGET: ${{ matrix.target }}
         run: |
-          set -eux
```
Comment (tico24) [inline .github/workflows/release.yaml:153]: I don't think this change was us. We should match upstream

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -150,97 +150,78 @@ jobs:
...
           tag=$(basename $GITHUB_REF)
           if [ $tag = "master" ]; then
             tag="latest"
           fi
-          # copied verbatim from Makefile
```
Comment (tico24) [inline .github/workflows/release.yaml]: And here

File: .github/workflows/release.yaml [hunk 5]
```diff
@@ -150,97 +150,78 @@ jobs:
...
+
+
           tag_suffix=$(echo $PLATFORM | sed -r "s/\//-/g")
           image_name="${DOCKERIO_ORG}/${TARGET}:${tag}-${tag_suffix}"
 
           docker buildx build \
             --cache-from "type=local,src=/tmp/.buildx-cache" \
             --cache-to "type=local,dest=/tmp/.buildx-cache" \
             --output "type=image,push=true" \
-            --build-arg GIT_COMMIT=$GIT_COMMIT \
```
Comment (tico24) [inline .github/workflows/release.yaml]: Here

File: Makefile [hunk 6]
```diff
@@ -14,12 +14,13 @@ GIT_BRANCH            := $(shell git rev-parse --symbolic-full-name --verify --q
 GIT_TAG               := $(shell git describe --exact-match --tags --abbrev=0  2> /dev/null || echo untagged)
 GIT_TREE_STATE        := $(shell if [ -z "`git status --porcelain`" ]; then echo "clean" ; else echo "dirty"; fi)
 RELEASE_TAG           := $(shell if [[ "$(GIT_TAG)" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.*$$ ]]; then echo "true"; else echo "false"; fi)
-DEV_BRANCH            := $(shell [ "$(GIT_BRANCH)" = master ] || [ `echo $(GIT_BRANCH) | cut -c -8` = release- ] || [ `echo $(GIT_BRANCH) | cut -c -4` = dev- ] || [ $(RELEASE_TAG) = true ] && echo false || echo true)
```
Comment (tico24) [inline Makefile]: Doubt this change was us.

**Review by tico24 — CHANGES_REQUESTED** [review]

File: Makefile (hunk 6 above)
Comment (JPZ13) [reply Makefile]: Yeah. I just matched the master branch for the upstream. Should I revert?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (tico24) [reply .github/workflows/release.yaml]: upstream is 0.10.4

File: Makefile (hunk 6 above)
Comment (tico24) [reply Makefile]: upstream has the double quotes. We should match upstream

**Comment by tico24** [comment]
@JPZ13 if you make the changes on a branch starting with `dev-` then you can do test releases.

**Comment by JPZ13** [comment]
Looks like we're pretty good on this. It should be the last blocker for cutting a release for @mostaphaRoudsari

@tico24 - could you take another look and cut the release if all is well?

**Review by tico24 — APPROVED** [review]

**Pull request merged**